src/test/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTaskTest.java

```diff
@@ -240,18 +256,40 @@ static boolean match(
     }
   }
 
-  private static final class InMemoryErrorReporter
-      implements Consumer<MongoProcessedSinkRecordData> {
-    private final List<MongoProcessedSinkRecordData> reported = new ArrayList<>();
+  private static final class InMemoryErrorReporter implements BiConsumer<SinkRecord, Exception> {
+    private final List<ReportedData> reported = new ArrayList<>();
 
     @Override
-    public void accept(final MongoProcessedSinkRecordData mongoProcessedSinkRecordData) {
-      reported.add(mongoProcessedSinkRecordData);
+    public void accept(final SinkRecord record, final Exception e) {
+      reported.add(new ReportedData(record, e));
     }
 
-    List<MongoProcessedSinkRecordData> reported() {
+    List<ReportedData> reported() {
       return unmodifiableList(reported);
     }
+
+    static final class ReportedData {
```

Comment:
This looks familiar to the `MongoProcessedSinkRecordData` - could we use a single data structure?

Reply:
`ReportedData` represents a pair of arguments passed to `StartedMongoSinkTask.errorReporter`. This pair of arguments is not the same thing as the pair `(MongoProcessedSinkRecordData.sinkRecord, MongoProcessedSinkRecordData.exception)`:
- It is semantically a different thing, as I mentioned [above](https://github.com/mongodb/mongo-kafka/pull/95#discussion_r779258628).
- `MongoProcessedSinkRecordData` does not provide a way to create an instance with an arbitrary exception. Refactoring this class to allow this does not make sense from the standpoint of the purpose of this class.

Reply:
Ack